.github/workflows/linux.yml

```diff
@@ -0,0 +1,87 @@
+name: Build Linux and openSBI in os-boot
+permissions:
+  id-token: write
+  attestations: write
+  contents: read
+
+on:
+  push:
+    paths:
+      - ".github/workflows/linux.yml"
+      - "os-boot/linux/**"
+      - "!os-boot/**.md"
+  pull_request:
+    paths:
+      - ".github/workflows/linux.yml"
+      - "os-boot/linux/**"
+      - "!os-boot/**.md"
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install packages
+        run: sudo apt install -y build-essential curl
+
+      - name: Check out repository code
+        uses: actions/checkout@v4
+
+      - name: Cache Sources
+        uses: actions/cache@v4
+        with:
+          key: ${{ hashFiles('os-boot/linux/Makefile') }}
+          path: os-boot/linux/src/*.tar.xz
+
+      - name: Make ELF
+        working-directory: os-boot/linux
```

Comment:
Should we use something like https://github.com/hendrikmuhs/ccache-action to speed up this build or cache the ELF file and only rebuild it on changes?

Reply:
Ah I missed this while looking at the file the first time. If I understand correctly, we will only rebuild linux if the makefile changes - is that correct?

Reply:
Yes because I put url and version number in it.

Reply:
Perfect, that sounds good to me.